governance/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,164 @@
+  - [Generic items we need to take into account:](#generic-items-we-need-to-take-into-account)
+- [Action Items](#action-items)
+- [Audit Requirements](#audit-requirements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+_Italics are used to indicate things that need to be replaced._
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | Julian Meyer                                       |
+| Created at         | 2025-02-10                                         |
+| Initial Reviewers  |                  |
+| Need Approval From |                            |
+| Status             | Draft |
+
+> [!NOTE]
+> 📢 Remember:
+>
+> - The single approver in the “Need Approval From” must be from the Security team.
+> - Maintain the “Status” property accordingly. An FMA document can have the following statuses:
+>   - **Draft 📝:** Doc is created but not yet ready for review.
+>   - **In Review 🔎:** Security is reviewing, and Engineering is iterating on the design. A checklist of action items will be created during this phase.
+>   - **Implementing Actions 🛫:** Security has signed off on the content of the document, including the resulting action items. Engineering is responsible for implementing the action items, and updating the checklist.
+>   - **Final 👍:** Security will transition the status of the document to Final once all action items are completed.
+
+> [!TIP]
+> Guidelines for writing a good analysis, and what the reviewer will look for:
+>
+> - Show your work: Include steps and tools for each conclusion.
+> - Completeness of risks considered.
+> - Include both implementation and operational failure modes
+> - Provide references to support the reviewer.
+> - The size of the document will likely be proportional to the project's complexity.
+> - The ultimate goal of this document is to identify action items to improve the security of the project. The FMA review process can be accelerated by proactively identifying action items during the writing process.
+
+## Introduction
+
+This document covers implementation of Pectra features in the OP stack. This requires a hard fork due to new transaction types like EIP-7702 transactions. This does not cover generic hardfork related security concerns, which will be addressed in a separate document.
+
+Included EIPs:
+
+-   EIP-7702: Set code transaction
+-   EIP-2537: BLS12-381 precompiles
+-   EIP-2935: Block hashes contract predeploy
+-   EIP-7251: Increase the MAX_EFFECTIVE_BALANCE
+-   EIP-7623: Increase calldata cost
+-   EIP-7002: Execution layer triggerable withdrawals
+-   EIP-6110: Supply validator deposits on chain
+-   EIP-7685: General purpose execution layer requests
```

Comment:
Are there OP Stack-specific considerations, design docs, specs, etc we can link to? This makes it seem like we are not doing any bespoke work and are just incorporating what e.g. upstream geth implements, and I am unsure if that is the case

Reply:
I added an explanation for each. There's no central source listing each change currently. We were going a lot off of this document: https://gist.github.com/protolambda/59b534d804c23dc7b3592ad1c38048d4

I can add this doc somewhere, but I think the updated explanations in this doc should give a good overview.